iree/compiler/Dialect/Flow/Transforms/CreateBenchmarkFuncs.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/compiler/Dialect/Flow/IR/FlowOps.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/BlockAndValueMapping.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/StandardTypes.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+
+// Creates wo kind of benchmark funcs:
+//   - Walks through all the execuatable ops and creates Funcs to invoke them.
+//   - Walks through all exported functions, create a func with dummy inputs,
+//     and explicitly remove export attribute from old functions.
+// The input are provided using flow.variable and flow.lookup.
+class CreateBenchmarkFuncs
+    : public PassWrapper<CreateBenchmarkFuncs, OperationPass<ModuleOp>> {
+ public:
+  CreateBenchmarkFuncs() = default;
+
+  void runOnOperation() override {
+    ModuleOp moduleOp = getOperation();
+    auto builder = OpBuilder::atBlockBegin(moduleOp.getBody());
+    SymbolTable moduleSymbols(moduleOp);
+    Location loc = builder.getUnknownLoc();
+    for (auto execOp : moduleOp.getOps<IREE::Flow::ExecutableOp>()) {
+      for (auto& op : execOp.getBlock()) {
+        auto dispatchEntryOp = dyn_cast<IREE::Flow::DispatchEntryOp>(op);
+        if (!dispatchEntryOp) continue;
+        auto execFuncOp = execOp.getInnerModule().lookupSymbol<FuncOp>(
+            dispatchEntryOp.function_ref());
+
+        // Create a funcOp to invoke the dispatch function.
+        std::string funcName = std::string(execFuncOp.getName()) + "_entry";
+        auto funcType =
+            builder.getFunctionType({}, execFuncOp.getType().getResults());
+        auto funcOp = builder.create<FuncOp>(loc, funcName, funcType);
+        funcOp.setAttr("iree.module.export", UnitAttr::get(&getContext()));
+        Block* block = funcOp.addEntryBlock();
+
+        // Build the body of the FuncOp.
+        OpBuilder::InsertionGuard gaurd(builder);
+        builder.setInsertionPoint(funcOp);
+        auto blockBuilder = OpBuilder(block, block->begin());
+        SmallVector<Value, 4> args;
+        for (auto inputType : execFuncOp.getType().getInputs()) {
+          args.push_back(
+              getDummyInput(builder, blockBuilder, inputType, moduleSymbols));
```

Comment:
Use the location of the argument

Reply:
I think using the location of the func op is enough? No need to extract the argument for just a location info.